File: src/main/java/net/fellter/more_leaf_particles/config/ConfigFields.java
package net.fellter.more_leaf_particles.config;

import dev.isxander.yacl3.config.v2.api.SerialEntry;
import org.jspecify.annotations.NonNull;

import java.awt.Color;

public class ConfigFields {
	@SerialEntry("enable_particle_rain_wind")
	public static boolean enableParticleRainWind = true;

	@SerialEntry("particle_rain_wind_multiplier")
	public static double particleRainWindMultiplier = 1.0F;

	@SerialEntry("ticks_to_blend")
	public static int ticksToBlend = 20;

	@SerialEntry("enable_oak")
	public static boolean enableOak = true;

	@SerialEntry("oak_multiplier")
	public static int oakMultiplier = 1;

	@SerialEntry("oak_size")
	public static float oakSize = 2.0F;

	@SerialEntry("oak_rotate")
	public static boolean oakRotate = true;

	@SerialEntry("enable_oak_custom_color")
	public static boolean enableOakCustomColor;

	@SerialEntry("oak_color")
	public static Color oakColor = new Color(0, 0, 0, 0);

	@SerialEntry("enable_oak_tint")
	public static boolean useOakTint = true;

	@SerialEntry("oak_gravity")
	public static float oakGravity = 0.07F;

	@SerialEntry("oak_wind")
	public static float oakWind = 10.0F;

	@SerialEntry("oak_flow_away")
	public static boolean oakFlowAway;

	@SerialEntry("oak_init_velocity")
	public static float oakInitialVelocity = 0.021F;

	@SerialEntry("enable_dark_oak")
	public static boolean enableDarkOak = true;

	@SerialEntry("dark_oak_multiplier")
	public static int darkOakMultiplier = 1;

	@SerialEntry("dark_oak_size")
	public static float darkOakSize = 2.0F;

	@SerialEntry("dark_oak_rotate")
	public static boolean darkOakRotate = true;

	@SerialEntry("enable_dark_oak_custom_color")
	public static boolean enableDarkOakCustomColor;

	@SerialEntry("dark_oak_color")
	@NonNull
	public static Color darkOakColor = new Color(0, 0, 0, 0);

	@SerialEntry("enable_dark_oak_tint")
	public static boolean useDarkOakTint = true;

	@SerialEntry("dark_oak_gravity")
	public static float darkOakGravity = 0.07F;

	@SerialEntry("dark_oak_wind")
	public static float darkOakWind = 10.0F;

	@SerialEntry("dark_oak_flow_away")
	public static boolean darkOakFlowAway;

	@SerialEntry("dark_oak_init_velocity")
	public static float darkOakInitialVelocity = 0.021F;

	@SerialEntry("enable_spruce")
	public static boolean enableSpruce = true;

	@SerialEntry("spruce_multiplier")
	public static int spruceMultiplier = 1;

	@SerialEntry("spruce_size")
	public static float spruceSize = 2.0F;

	@SerialEntry("spruce_rotate")
	public static boolean spruceRotate = true;

	@SerialEntry("enable_spruce_custom_color")
	public static boolean enableSpruceCustomColor;

	@SerialEntry("spruce_color")
	@NonNull
	public static Color spruceColor = new Color(0, 0, 0, 0);

	@SerialEntry("enable_spruce_tint")
	public static boolean useSpruceTint = true;

	@SerialEntry("spruce_gravity")
	public static float spruceGravity = 0.07F;

	@SerialEntry("spruce_wind")
	public static float spruceWind = 10.0F;

	@SerialEntry("spruce_flow_away")
	public static boolean spruceFlowAway;

	@SerialEntry("spruce_init_velocity")
	public static float spruceInitialVelocity = 0.021F;

	@SerialEntry("enable_birch")
	public static boolean enableBirch = true;

	@SerialEntry("birch_multiplier")
	public static int birchMultiplier = 1;

	@SerialEntry("birch_size")
	public static float birchSize = 2.0F;

	@SerialEntry("birch_rotate")
	public static boolean birchRotate = true;

	@SerialEntry("enable_birch_custom_color")
	public static boolean enableBirchCustomColor;

	@SerialEntry("birch_color")
	@NonNull
	public static Color birchColor = new Color(0, 0, 0, 0);

	@SerialEntry("enable_birch_tint")
	public static boolean useBirchTint = true;

	@SerialEntry("birch_gravity")
	public static float birchGravity = 0.07F;

	@SerialEntry("birch_wind")
	public static float birchWind = 10.0F;

	@SerialEntry("birch_flow_away")
	public static boolean birchFlowAway;

	@SerialEntry("birch_init_velocity")
	public static float birchInitialVelocity = 0.021F;

	@SerialEntry("enable_mangrove")
	public static boolean enableMangrove = true;

	@SerialEntry("mangrove_multiplier")
	public static int mangroveMultiplier = 1;

	@SerialEntry("mangrove_size")
	public static float mangroveSize = 2.0F;

	@SerialEntry("mangrove_rotate")
	public static boolean mangroveRotate = true;

	@SerialEntry("enable_mangrove_custom_color")
	public static boolean enableMangroveCustomColor;

	@SerialEntry("mangrove_color")
	@NonNull
	public static Color mangroveColor = new Color(0, 0, 0, 0);

	@SerialEntry("enable_mangrove_tint")
	public static boolean useMangroveTint = true;

	@SerialEntry("mangrove_gravity")
	public static float mangroveGravity = 0.07F;

	@SerialEntry("mangrove_wind")
	public static float mangroveWind = 10.0F;

	@SerialEntry("mangrove_flow_away")
	public static boolean mangroveFlowAway;

	@SerialEntry("mangrove_init_velocity")
	public static float mangroveInitialVelocity = 0.021F;

	@SerialEntry("enable_jungle")
	public static boolean enableJungle = true;

	@SerialEntry("jungle_multiplier")
	public static int jungleMultiplier = 1;

	@SerialEntry("jungle_size")
	public static float jungleSize = 2.0F;

	@SerialEntry("jungle_rotate")
	public static boolean jungleRotate = true;

	@SerialEntry("enable_jungle_custom_color")
	public static boolean enableJungleCustomColor;

	@SerialEntry("jungle_color")
	@NonNull
	public static Color jungleColor = new Color(0, 0, 0, 0);

	@SerialEntry("enable_jungle_tint")
	public static boolean useJungleTint = true;

	@SerialEntry("jungle_gravity")
	public static float jungleGravity = 0.07F;

	@SerialEntry("jungle_wind")
	public static float jungleWind = 10.0F;

	@SerialEntry("jungle_flow_away")
	public static boolean jungleFlowAway;

	@SerialEntry("jungle_init_velocity")
	public static float jungleInitialVelocity = 0.021F;

	@SerialEntry("enable_acacia")
	public static boolean enableAcacia = true;

	@SerialEntry("acacia_multiplier")
	public static int acaciaMultiplier = 1;

	@SerialEntry("acacia_size")
	public static float acaciaSize = 1.66F;

	@SerialEntry("acacia_rotate")
	public static boolean acaciaRotate = true;

	@SerialEntry("enable_acacia_custom_color")
	public static boolean enableAcaciaCustomColor;

	@SerialEntry("acacia_color")
	@NonNull
	public static Color acaciaColor = new Color(0, 0, 0, 0);

	@SerialEntry("enable_acacia_tint")
	public static boolean useAcaciaTint = true;

	@SerialEntry("acacia_gravity")
	public static float acaciaGravity = 0.07F;

	@SerialEntry("acacia_wind")
	public static float acaciaWind = 10.0F;

	@SerialEntry("acacia_flow_away")
	public static boolean acaciaFlowAway;

	@SerialEntry("acacia_init_velocity")
	public static float acaciaInitialVelocity = 0.021F;

	@SerialEntry("enable_azalea")
	public static boolean enableAzalea = true;

	@SerialEntry("azalea_multiplier")
	public static int azaleaMultiplier = 1;

	@SerialEntry("azalea_size")
	public static float azaleaSize = 2.0F;

	@SerialEntry("azalea_rotate")
	public static boolean azaleaRotate = true;

	@SerialEntry("azalea_gravity")
	public static float azaleaGravity = 0.07F;

	@SerialEntry("azalea_wind")
	public static float azaleaWind = 10.0F;

	@SerialEntry("azalea_flow_away")
	public static boolean azaleaFlowAway;

	@SerialEntry("azalea_init_velocity")
	public static float azaleaInitialVelocity = 0.021F;

	@SerialEntry("enable_pale_oak")
	public static boolean enablePaleOak = true;

	@SerialEntry("pale_oak_multiplier")
	public static int paleOakMultiplier = 1;

	@SerialEntry("pale_oak_size")
	public static float paleOakSize = 2.0F;

	@SerialEntry("pale_oak_rotate")
	public static boolean paleOakRotate = true;

	@SerialEntry("pale_oak_gravity")
	public static float paleOakGravity = 0.07F;

	@SerialEntry("pale_oak_wind")
	public static float paleOakWind = 10.0F;

	@SerialEntry("pale_oak_flow_away")
	public static boolean paleOakFlowAway;

	@SerialEntry("pale_oak_init_velocity")
	public static float paleOakInitialVelocity = 0.021F;

	@SerialEntry("enable_cherry")
	public static boolean enableCherry = true;

	@SerialEntry("cherry_multiplier")
	public static int cherryMultiplier = 1;

	@SerialEntry("cherry_size")
	public static float cherrySize = 2.0F;

	@SerialEntry("cherry_rotate")
	public static boolean cherryRotate = true;

	@SerialEntry("cherry_gravity")
	public static float cherryGravity = 0.07F;

	@SerialEntry("cherry_wind")
	public static float cherryWind = 10.0F;

	@SerialEntry("cherry_flow_away")
	public static boolean cherryFlowAway;

	@SerialEntry("cherry_init_velocity")
	public static float cherryInitialVelocity = 0.021F;

	@SerialEntry("enable_flowering_azalea")
	public static boolean enableFloweringAzalea = true;

	@SerialEntry("flowering_azalea_multiplier")
	public static int floweringAzaleaMultiplier = 1;

	@SerialEntry("flowering_azalea_size")
	public static float floweringAzaleaSize = 2.0F;

	@SerialEntry("flowering_azalea_rotate")
	public static boolean floweringAzaleaRotate = true;

	@SerialEntry("flowering_azalea_gravity")
	public static float floweringAzaleaGravity = 0.07F;

	@SerialEntry("flowering_azalea_wind")
	public static float floweringAzaleaWind = 10.0F;

	@SerialEntry("flowering_azalea_flow_away")
	public static boolean floweringAzaleaFlowAway;

	@SerialEntry("flowering_azalea_init_velocity")
	public static float floweringAzaleaInitialVelocity = 0.021F;

}

File: src/main/java/net/fellter/moreLeafParticles/config/ConfigFields.kt
package net.fellter.moreLeafParticles.config

import dev.isxander.yacl3.config.v2.api.SerialEntry
import java.awt.Color

/*
 * Has to be a class instead of an object because YACL relies on
 * retrieving a no-arg primary constructor, and objects do not have one.
 */
class ConfigFields {
	companion object {
		/* COMMON */
		@JvmField @SerialEntry("enable_particle_rain_wind")
		var enableParticleRainWind: Boolean = true

		@JvmField @SerialEntry("particle_rain_wind_multiplier")
		var particleRainWindMultiplier: Double = 1.0

		@JvmField @SerialEntry("ticks_to_blend")
		var ticksToBlend: Int = 20

		/* OAK */
		@JvmField @SerialEntry(value = "enable_oak")
		var enableOak: Boolean = true

		@JvmField @SerialEntry(value = "oak_multiplier")
		var oakMultiplier: Int = 1

		@JvmField @SerialEntry(value = "oak_size")
		var oakSize: Float = 2.0f

		@JvmField @SerialEntry(value = "oak_rotate")
		var oakRotate: Boolean = true

		@JvmField @SerialEntry(value = "enable_oak_custom_color")
		var enableOakCustomColor: Boolean = false

		@JvmField @SerialEntry(value = "oak_color")
		var oakColor: Color = Color(0, 0, 0, 0)

		@JvmField @SerialEntry(value = "enable_oak_tint")
		var useOakTint: Boolean = true

		@JvmField @SerialEntry(value = "oak_gravity")
		var oakGravity: Float = 0.07f

		@JvmField @SerialEntry(value = "oak_wind")
		var oakWind: Float = 10.0f

		@JvmField @SerialEntry(value = "oak_flow_away")
		var oakFlowAway: Boolean = false

		@JvmField @SerialEntry(value = "oak_init_velocity")
		var oakInitialVelocity: Float = 0.021f

		/* DARK OAK */
		@JvmField @SerialEntry(value = "enable_dark_oak")
		var enableDarkOak: Boolean = true

		@JvmField @SerialEntry(value = "dark_oak_multiplier")
		var darkOakMultiplier: Int = 1

		@JvmField @SerialEntry(value = "dark_oak_size")
		var darkOakSize: Float = 2.0f

		@JvmField @SerialEntry(value = "dark_oak_rotate")
		var darkOakRotate: Boolean = true

		@JvmField @SerialEntry(value = "enable_dark_oak_custom_color")
		var enableDarkOakCustomColor: Boolean = false

		@JvmField @SerialEntry(value = "dark_oak_color")
		var darkOakColor: Color = Color(0, 0, 0, 0)

		@JvmField @SerialEntry(value = "enable_dark_oak_tint")
		var useDarkOakTint: Boolean = true

		@JvmField @SerialEntry(value = "dark_oak_gravity")
		var darkOakGravity: Float = 0.07f

		@JvmField @SerialEntry(value = "dark_oak_wind")
		var darkOakWind: Float = 10f

		@JvmField @SerialEntry(value = "dark_oak_flow_away")
		var darkOakFlowAway: Boolean = false

		@JvmField @SerialEntry(value = "dark_oak_init_velocity")
		var darkOakInitialVelocity: Float = 0.021f

		/* SPRUCE */
		@JvmField @SerialEntry(value = "enable_spruce")
		var enableSpruce: Boolean = true

		@JvmField @SerialEntry(value = "spruce_multiplier")
		var spruceMultiplier: Int = 1

		@JvmField @SerialEntry(value = "spruce_size")
		var spruceSize: Float = 2.0f

		@JvmField @SerialEntry(value = "spruce_rotate")
		var spruceRotate: Boolean = true

		@JvmField @SerialEntry(value = "enable_spruce_custom_color")
		var enableSpruceCustomColor: Boolean = false

		@JvmField @SerialEntry(value = "spruce_color")
		var spruceColor: Color = Color(0, 0, 0, 0)

		@JvmField @SerialEntry(value = "enable_spruce_tint")
		var useSpruceTint: Boolean = true

		@JvmField @SerialEntry(value = "spruce_gravity")
		var spruceGravity: Float = 0.07f

		@JvmField @SerialEntry(value = "spruce_wind")
		var spruceWind: Float = 10f

		@JvmField @SerialEntry(value = "spruce_flow_away")
		var spruceFlowAway: Boolean = false

		@JvmField @SerialEntry(value = "spruce_init_velocity")
		var spruceInitialVelocity: Float = 0.021f

		/* BIRCH */
		@JvmField @SerialEntry(value = "enable_birch")
		var enableBirch: Boolean = true

		@JvmField @SerialEntry(value = "birch_multiplier")
		var birchMultiplier: Int = 1

		@JvmField @SerialEntry(value = "birch_size")
		var birchSize: Float = 2.0f

		@JvmField @SerialEntry(value = "birch_rotate")
		var birchRotate: Boolean = true

		@JvmField @SerialEntry(value = "enable_birch_custom_color")
		var enableBirchCustomColor: Boolean = false

		@JvmField @SerialEntry(value = "birch_color")
		var birchColor: Color = Color(0, 0, 0, 0)

		@JvmField @SerialEntry(value = "enable_birch_tint")
		var useBirchTint: Boolean = true

		@JvmField @SerialEntry(value = "birch_gravity")
		var birchGravity: Float = 0.07f

		@JvmField @SerialEntry(value = "birch_wind")
		var birchWind: Float = 10f

		@JvmField @SerialEntry(value = "birch_flow_away")
		var birchFlowAway: Boolean = false

		@JvmField @SerialEntry(value = "birch_init_velocity")
		var birchInitialVelocity: Float = 0.021f

		/* MANGROVE */
		@JvmField @SerialEntry(value = "enable_mangrove")
		var enableMangrove: Boolean = true

		@JvmField @SerialEntry(value = "mangrove_multiplier")
		var mangroveMultiplier: Int = 1

		@JvmField @SerialEntry(value = "mangrove_size")
		var mangroveSize: Float = 2.0f

		@JvmField @SerialEntry(value = "mangrove_rotate")
		var mangroveRotate: Boolean = true

		@JvmField @SerialEntry(value = "enable_mangrove_custom_color")
		var enableMangroveCustomColor: Boolean = false

		@JvmField @SerialEntry(value = "mangrove_color")
		var mangroveColor: Color = Color(0, 0, 0, 0)

		@JvmField @SerialEntry(value = "enable_mangrove_tint")
		var useMangroveTint: Boolean = true

		@JvmField @SerialEntry(value = "mangrove_gravity")
		var mangroveGravity: Float = 0.07f

		@JvmField @SerialEntry(value = "mangrove_wind")
		var mangroveWind: Float = 10f

		@JvmField @SerialEntry(value = "mangrove_flow_away")
		var mangroveFlowAway: Boolean = false

		@JvmField @SerialEntry(value = "mangrove_init_velocity")
		var mangroveInitialVelocity: Float = 0.021f

		/* JUNGLE */
		@JvmField @SerialEntry(value = "enable_jungle")
		var enableJungle: Boolean = true

		@JvmField @SerialEntry(value = "jungle_multiplier")
		var jungleMultiplier: Int = 1

		@JvmField @SerialEntry(value = "jungle_size")
		var jungleSize: Float = 2.0f

		@JvmField @SerialEntry(value = "jungle_rotate")
		var jungleRotate: Boolean = true

		@JvmField @SerialEntry(value = "enable_jungle_custom_color")
		var enableJungleCustomColor: Boolean = false

		@JvmField @SerialEntry(value = "jungle_color")
		var jungleColor: Color = Color(0, 0, 0, 0)

		@JvmField @SerialEntry(value = "enable_jungle_tint")
		var useJungleTint: Boolean = true

		@JvmField @SerialEntry(value = "jungle_gravity")
		var jungleGravity: Float = 0.07f

		@JvmField @SerialEntry(value = "jungle_wind")
		var jungleWind: Float = 10f

		@JvmField @SerialEntry(value = "jungle_flow_away")
		var jungleFlowAway: Boolean = false

		@JvmField @SerialEntry(value = "jungle_init_velocity")
		var jungleInitialVelocity: Float = 0.021f

		/* ACACIA */
		@JvmField @SerialEntry(value = "enable_acacia")
		var enableAcacia: Boolean = true

		@JvmField @SerialEntry(value = "acacia_multiplier")
		var acaciaMultiplier: Int = 1

		@JvmField @SerialEntry(value = "acacia_size")
		var acaciaSize: Float = 1.66f

		@JvmField @SerialEntry(value = "acacia_rotate")
		var acaciaRotate: Boolean = true

		@JvmField @SerialEntry(value = "enable_acacia_custom_color")
		var enableAcaciaCustomColor: Boolean = false

		@JvmField @SerialEntry(value = "acacia_color")
		var acaciaColor: Color = Color(0, 0, 0, 0)

		@JvmField @SerialEntry(value = "enable_acacia_tint")
		var useAcaciaTint: Boolean = true

		@JvmField @SerialEntry(value = "acacia_gravity")
		var acaciaGravity: Float = 0.07f

		@JvmField @SerialEntry(value = "acacia_wind")
		var acaciaWind: Float = 10f

		@JvmField @SerialEntry(value = "acacia_flow_away")
		var acaciaFlowAway: Boolean = false

		@JvmField @SerialEntry(value = "acacia_init_velocity")
		var acaciaInitialVelocity: Float = 0.021f

		/* AZALEA */
		@JvmField @SerialEntry(value = "enable_azalea")
		var enableAzalea: Boolean = true

		@JvmField @SerialEntry(value = "azalea_multiplier")
		var azaleaMultiplier: Int = 1

		@JvmField @SerialEntry(value = "azalea_size")
		var azaleaSize: Float = 2.0f

		@JvmField @SerialEntry(value = "azalea_rotate")
		var azaleaRotate: Boolean = true

		@JvmField @SerialEntry(value = "azalea_gravity")
		var azaleaGravity: Float = 0.07f

		@JvmField @SerialEntry(value = "azalea_wind")
		var azaleaWind: Float = 10f

		@JvmField @SerialEntry(value = "azalea_flow_away")
		var azaleaFlowAway: Boolean = false

		@JvmField @SerialEntry(value = "azalea_init_velocity")
		var azaleaInitialVelocity: Float = 0.021f

		/* PALE OAK */
		@JvmField @SerialEntry(value = "enable_pale_oak")
		var enablePaleOak: Boolean = true

		@JvmField @SerialEntry(value = "pale_oak_multiplier")
		var paleOakMultiplier: Int = 1

		@JvmField @SerialEntry(value = "pale_oak_size")
		var paleOakSize: Float = 2.0f

		@JvmField @SerialEntry(value = "pale_oak_rotate")
		var paleOakRotate: Boolean = true

		@JvmField @SerialEntry(value = "pale_oak_gravity")
		var paleOakGravity: Float = 0.07f

		@JvmField @SerialEntry(value = "pale_oak_wind")
		var paleOakWind: Float = 10f

		@JvmField @SerialEntry(value = "pale_oak_flow_away")
		var paleOakFlowAway: Boolean = false

		@JvmField @SerialEntry(value = "pale_oak_init_velocity")
		var paleOakInitialVelocity: Float = 0.021f

		/* CHERRY */
		@JvmField @SerialEntry(value = "enable_cherry")
		var enableCherry: Boolean = true

		@JvmField @SerialEntry(value = "cherry_multiplier")
		var cherryMultiplier: Int = 1

		@JvmField @SerialEntry(value = "cherry_size")
		var cherrySize: Float = 2.0f

		@JvmField @SerialEntry(value = "cherry_rotate")
		var cherryRotate: Boolean = true

		@JvmField @SerialEntry(value = "cherry_gravity")
		var cherryGravity: Float = 0.07f

		@JvmField @SerialEntry(value = "cherry_wind")
		var cherryWind: Float = 10f

		@JvmField @SerialEntry(value = "cherry_flow_away")
		var cherryFlowAway: Boolean = false

		@JvmField @SerialEntry(value = "cherry_init_velocity")
		var cherryInitialVelocity: Float = 0.021f

		/* FLOWERING AZALEA */
		@JvmField @SerialEntry(value = "enable_flowering_azalea")
		var enableFloweringAzalea: Boolean = true

		@JvmField @SerialEntry(value = "flowering_azalea_multiplier")
		var floweringAzaleaMultiplier: Int = 1

		@JvmField @SerialEntry(value = "flowering_azalea_size")
		var floweringAzaleaSize: Float = 2.0f

		@JvmField @SerialEntry(value = "flowering_azalea_rotate")
		var floweringAzaleaRotate: Boolean = true

		@JvmField @SerialEntry(value = "flowering_azalea_gravity")
		var floweringAzaleaGravity: Float = 0.07f

		@JvmField @SerialEntry(value = "flowering_azalea_wind")
		var floweringAzaleaWind: Float = 10f

		@JvmField @SerialEntry(value = "flowering_azalea_flow_away")
		var floweringAzaleaFlowAway: Boolean = false

		@JvmField @SerialEntry(value = "flowering_azalea_init_velocity")
		var floweringAzaleaInitialVelocity: Float = 0.021f
	}
}